### PR TITLE
Upgrade to ODFE 1.12.0

### PR DESCRIPTION
**Pull request by joshuali925**
*Issue #, if available:*

*Description of changes:*
- Compatible with Kibana 7.10.0
- Add github actions
- minor refactor

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: .github/workflows/release-workflow.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+          node-version: '10.22.1'
+
+      - name: Kibana Pluign Bootstrap
+        run: |
+          yarn kbn bootstrap
+
+      - name: Build Artifact
+        run: |
+          yarn build
+          mv ./build/*.zip ./build/opendistroTraceAnalyticsKibana-1.12.0.0.zip
```
Comment (zhongnansu) [inline .github/workflows/release-workflow.yml:49]: maybe use a env variable for version number and name?

File: .kibana-plugin-helpers.json [hunk 2]
```diff
@@ -1,11 +1,12 @@
...
+  "serverSourcePatterns": [
     "package.json",
     "yarn.lock",
     "tsconfig.json",
-    "{public,server}/**/*",
-    "kibana.json",
+    "common/**/*",
+    "public/**/*",
+    "server/**/*",
     "target/**/*",
```
Comment (zhongnansu) [inline .kibana-plugin-helpers.json:9]: do we still need this？

File: package.json [hunk 3]
```diff
@@ -1,46 +1,31 @@
...
-    "start": "plugin-helpers start",
-    "test:server": "plugin-helpers test:server",
-    "test:browser": "plugin-helpers test:browser",
-    "build": "plugin-helpers build"
-  },
   "devDependencies": {
-    "@elastic/eslint-config-kibana": "link:../../packages/eslint-config-kibana",
-    "@elastic/eslint-import-resolver-kibana": "link:../../packages/kbn-eslint-import-resolver-kibana",
-    "@kbn/plugin-helpers": "link:../../packages/kbn-plugin-helpers",
+    "cypress": "^5.0.0",
```
Comment (zhongnansu) [inline package.json:23]: Np: I think we can get rid of this as well, and use the cypress in Kibana, need to change the script for running cypress as well

File: release-notes/opendistro-for-elasticsearch-trace-analytics.release-notes-1.12.0.0.md [hunk 4]
```diff
@@ -0,0 +1,12 @@
+## 2020-11-18 Version 1.12.0.0
+
+### Features
+* Rebase trace analytics to main branch ([#1](https://github.com/opendistro-for-elasticsearch/trace-analytics/pull/1))
```
Comment (zhongnansu) [inline release-notes/opendistro-for-elasticsearch-trace-analytics.release-notes-1.12.0.0.md:4]: just curious, is this manually added or auto-generated

**Review by zhongnansu — APPROVED** [review]
left few comments

File: .github/workflows/release-workflow.yml (hunk 1 above)
Comment (joshuali925) [reply .github/workflows/release-workflow.yml:49]: good idea, will update

File: .kibana-plugin-helpers.json (hunk 2 above)
Comment (zhongnansu) [reply .kibana-plugin-helpers.json:9]: I think the new `kbn-plugin-helper` can include the target in the artifact, so no need to specify here. I verified with reporting. You can also give it a try

Comment (joshuali925) [reply .kibana-plugin-helpers.json]: you are right it doesn't matter now, removed

File: release-notes/opendistro-for-elasticsearch-trace-analytics.release-notes-1.12.0.0.md (hunk 4 above)
Comment (joshuali925) [reply release-notes/opendistro-for-elasticsearch-trace-analytics.release-notes-1.12.0.0.md:4]: I'm doing it manually which doesn't take much time using macros, but there's also github [api](https://api.github.com/repos/opendistro-for-elasticsearch/trace-analytics/pulls?state=closed) or [release-drafter](https://github.com/release-drafter/release-drafter)

**Pull request merged**